### ISSUE TITLE
OLM installation broken when having cert-controller/webhook disabled

### ISSUE THREAD

**Issue by slopezz** [external-secrets/external-secrets#1063]
**Describe the solution you'd like**

We were using ESO **v0.5.1** installed via OLM, where we previously had disabled cert-controller and webhook for different reasons:
- They seemed to be optional (although enabled by default)
- **We already updated our `ExternalSecrets` and `ClusterSecretsStores` to `v1beta1` (so nothing in `v1alpha1`, so no need to convert anything)**
- Since the upgrade to **v0.5.X** (with cert-controller/webhook enabled by default), our Vault backend (AWS DynamoDB) increased the read/write operations by 4x (which means we were paying 4x the cost of AWS dynamodb), so we disabled both cert-controller/webhook and ESO was working OK, and AWS DynamoDB read/write operations reduced again to ESO **v0.4.X** values

However, tonight a few of our clusters with automatic updates have tried to upgrade ESO from **v0.5.1** to **v0.5.2**, and all of them failed, because the it seems the upgrade requires to have the webhook `Service` running (so the webhook is previously enabled):

```
Failed
error validating existing CRs against new CRD's schema for "clustersecretstores.external-secrets.io": error listing resources in GroupVersionResource schema.GroupVersionResource{Group:"external-secrets.io", Version:"v1alpha1", Resource:"clustersecretstores"}: conversion webhook for external-secrets.io/v1beta1, Kind=ClusterSecretStore failed: Post "https://cluster-external-secrets-webhook.external-secrets.svc:443/convert?timeout=30s": service "cluster-external-secrets-webhook" not found
```

So it seems the upgrade requires to have previously the `Service` `cluster-external-secrets-webhook`, which only exists if webhook is enabled.

We tried to start ESO installation from scratch on a cluster, by unistalling ESO operator and install it again with a fresh OLM Subscription (this operation does not break current `ExternalSecrets` and their associated `Secrets`), but the fresh install also fails because it expects to have the webhook service, which of course does not exist on a fresh install.

So it seems there is a **chicken-egg problem**.  Why a fresh install (or an upgrade) requires an specific webhook `Service` to be present?
- On a fresh operator install is impossible to have it present
- On an operator upgrade, only if cert-controller/webhook is runnign before (so enabled), specific webhook `Service` wiull be present and upgrade will succeed

**Workaround on affected cluster whose previous ESO is still running and cert-controller/webhook are disabled**

We mitigated the upgrade on other affected clusters where **0.5.2** upgrade failed (but **0.5.1** was still running, we had not yet unistalled it), by enabling cert-controller/webhook first (running ESO **0.5.1**  will create them), so that way unistall the operator and then install the operator again it, because the webhook `Service` already is present thanks to have enabled it just before the upgrade.

**Possible workaround on affected cluster where ESO is not running and cert-controller/webhook are disabled**

For the cluster which is failing and we cannot do anything because ESO was unistalled without having before creating the webhook, we think we might "workaround it" by **manually** creating the expected cert-controller/webhook k8s resources, so when the install is being done by OLM, the required webhook `Service` will be there.

**Comment by gusfcarvalho** [comment]
Hi @slopezz ! Another workaround for your issue would be to setup CRDs like this (although, I'm not sure if that is applicable in your use-case):
```
conversion:
  strategy: None
```

The `webhook` pod is not meant to be optional, though. It contains the ValidatingWebhooks for ExternalSecrets and for SecretStores, effectively blocking some misconfiguration that could be potentially dangerous to operations. The recommendation is to have one webhook deployment per cluster.

The option `create` it is to allow multiple external-secrets deployments (with scoped namespaces) but only one webhook deployment.

certController is optional, as the only job it is doing is providing valid certificate credentials to CRDs, ValidatingWebhooks and the webhook pod itself. It can be replaced with cert-manager. It should be only one per cluster, just like the webhook.


The issue is actually why with certController and webhook (which are passive components) we get an increase of the requests to Hashicorp vault (we've seen that before if you have both v0.4.x version in conjunction with v0.5.x version, as each version will store a different eso `apiVersion`, leading to a much faster reconcile than intended).

**Comment by slopezz** [comment]
Thanks @gusfcarvalho for the feedback

Knowing webhook cannot be optional, we have just re-enabled again (along with cert-controller) on all our 5 clusters, and everything is running OK now.

Vault read-writes are stable now, when we notice the 4x increase on DynamoDB read/writes operations we had a mix of ESO `v0.4.X` and `v0.5.X`, exactly what you mention as a kind of possible root cause, but now we only use `v0.5.X`:
![image](https://user-images.githubusercontent.com/41513123/166463741-54a2fea9-dd9a-4638-a9a6-e1db5953b2f1.png)

On the cluster where operator was already unistalled without having the webhook enabled, I just created the needed resources manually and now the operator installation is successful without issues. However although OLM installation was successful, there was a few errors like `Unable to continue with update: ServiceAccount \"X\" in namespace \"X\" exists and cannot be imported into the current release: invalid ownership metadata...` (normal, they were not created by the operator itself, were created manually by me), so I just removed these manual resources and also the `OperatorConfig` resource, and created `OperatorConfig` back again, now everything has been recreated correctly.

So I proceed to close the issue, than you for the support!
